plugins/GSdx/Renderers/DX11/GSDevice11.h

```diff
@@ -21,27 +21,315 @@
+		union
+		{
+			struct
+			{
+				// *** Word 1
+				// Format
+				uint32 fmt:4;
+				uint32 dfmt:2;
+				uint32 depth_fmt:2;
+				// Alpha extension/Correction
+				uint32 aem:1;
+				uint32 fba:1;
+				// Fog
+				uint32 fog:1;
+				// Pixel test
+				uint32 atst:3;
+				// Color sampling
+				uint32 fst:1;
+				uint32 tfx:3;
+				uint32 tcc:1;
+				uint32 wms:2;
+				uint32 wmt:2;
+				uint32 ltf:1;
+				// Shuffle and fbmask effect
+				uint32 shuffle:1;
+				uint32 read_ba:1;
+
+				// *** Word 2
+				// Blend and Colclip
+				uint32 clr1:1;
+
+				// Others ways to fetch the texture
+				uint32 channel:3;
+
+				// Hack
+				uint32 aout:1;
+				uint32 spritehack:1;
+				uint32 tcoffsethack:1;
+				uint32 urban_chaos_hle:1;
+				uint32 tales_of_abyss_hle:1;
+				uint32 point_sampler:1;
+
+				uint32 _free:28;
+			};
+
+			uint64 key;
+		};
+
+		operator uint64() {return key;}
+
+		PSSelector() : key(0) {}
+	};
+
+	struct PSSamplerSelector
+	{
+		union
+		{
+			struct
+			{
+				uint32 tau:1;
+				uint32 tav:1;
+				uint32 ltf:1;
+			};
+
+			uint32 key;
+		};
+
+		operator uint32() {return key & 0x7;}
+
+		PSSamplerSelector() : key(0) {}
+	};
+
+	struct OMDepthStencilSelector
+	{
+		union
+		{
+			struct
+			{
+				uint32 ztst:2;
+				uint32 zwe:1;
+				uint32 date:1;
+				uint32 fba:1;
+				uint32 date_one:1;
+			};
+
+			uint32 key;
+		};
+
+		operator uint32() {return key & 0x3f;}
+
+		OMDepthStencilSelector() : key(0) {}
+	};
+
+	struct OMBlendSelector
+	{
+		union
+		{
+			struct
+			{
+				uint32 abe:1;
+				uint32 a:2;
+				uint32 b:2;
+				uint32 c:2;
+				uint32 d:2;
+				uint32 wr:1;
+				uint32 wg:1;
+				uint32 wb:1;
+				uint32 wa:1;
+			};
+
+			struct
+			{
+				uint32 _pad:1;
+				uint32 abcd:8;
+				uint32 wrgba:4;
+			};
+
+			uint32 key;
+		};
+
+		operator uint32() {return key & 0x1fff;}
+
+		OMBlendSelector() : key(0) {}
+
+		bool IsCLR1() const
+		{
+			return (key & 0x19f) == 0x93; // abe == 1 && a == 1 && b == 2 && d == 1
+		}
+	};
+
+	struct D3D11Blend
+	{
+		int bogus;
+		D3D11_BLEND_OP op;
+		D3D11_BLEND src, dst;
+	};
+	static const D3D11Blend m_blendMapD3D11[3*3*3*3];
+
+	#pragma pack(pop)
+private:
+	float m_hack_topleft_offset;
+	int m_upscale_multiplier;
+	int m_mipmap;
+
 	GSTexture* CreateSurface(int type, int w, int h, int format);
+	GSTexture* FetchSurface(int type, int w, int h, int format);
 
-	void DoMerge(GSTexture* sTex[3], GSVector4* sRect, GSTexture* dTex, GSVector4* dRect, const GSRegPMODE& PMODE, const GSRegEXTBUF& EXTBUF, const GSVector4& c);
-	void DoInterlace(GSTexture* sTex, GSTexture* dTex, int shader, bool linear, float yoffset = 0);
-	void DoFXAA(GSTexture* sTex, GSTexture* dTex);
-	void DoShadeBoost(GSTexture* sTex, GSTexture* dTex);
-	void DoExternalFX(GSTexture* sTex, GSTexture* dTex);
-	void RenderOsd(GSTexture* dt);
+	void DoMerge(GSTexture* sTex[3], GSVector4* sRect, GSTexture* dTex, GSVector4* dRect, const GSRegPMODE& PMODE, const GSRegEXTBUF& EXTBUF, const GSVector4& c) final;
```

Comment:
For the record, Final is useless if the class is already final. Don't remember what I did for gl